lisp/ess-r-package.el

```diff
@@ -268,6 +273,24 @@ arguments, or expressions which return R arguments."
         args
       (concat ", " args))))
 
+(defun ess-r-package-compilation-start (command)
+  "Run `compilation-start' in the package's root directory."
+  (let* ((pkg-info (or (ess-r-package-project)
+                       (ess-r-package-set-package)))
+         (pkg-name (ess-r-package-name))
+         (default-directory (cdr pkg-info)))
+    (compilation-start command nil
```

Comment:
`t` instead of nil would result in compilation-shell-minor-mode which should solve the term signals issue.

Reply:
Unfortunately this also puts it in `comint-mode`, which is quite a different experience than the usual `compilation-mode`. I don't see an obvious way to get term signal handling outside of `comint-mode` -- do you have any inkling of what could be done?

Reply:
Why does that matter? If compilation mode is based on comint this is how it is.